src/search-field.tsx

```diff
@@ -16,6 +16,7 @@ export interface SearchFieldProps extends CommonFormFieldProps {
     onChangeValue?: (value: string, rawValue: string) => void;
     getSuggestions?: (value: string) => ReadonlyArray<string>;
     inputMode?: React.HTMLAttributes<HTMLInputElement>['inputMode'];
+    withSearchIcon?: boolean;
```

Comment:
maybe is more scalable called "hideStartIcon"? This name could be the same in any field in the future if we decide to open this config

Reply:
my initial idea was `startIcon` to allow customizing it (allowing null/undefined for removing it). But @aweell told me we don't want that customization.

Following your proposal I think I prefer something like `withStartIcon`. I prefer positive booleans, even if the default value is true